### DISCUSSION TITLE
When using proxy remote url, can I make the registry validate the checksum when pulling an image?

### DISCUSSION THREAD

**Discussion by amir-bialek** [distribution/distribution#4455] [General]
In our k8s on prem we have a local image registry from image [docker.io/registry:2.8.3](http://docker.io/registry:2.8.3) that is a proxy from our dockerhub (registry mirror).
Sometimes our self-hosted-runners (ARC) get an error when trying to pull an image from it:
```
ERROR: failed to solve: failed to read expected number of bytes: unexpected EOF
```
This issue occurs when the manifest in the local registry is corrupted. a quick-fix we do is delete the image from our local registry and re-run the job. this cause the local registry to re-pull it from [docker.io](http://docker.io/)

I will like to find a permanent solution, will appreciate your help.

Is it possible to make the registry always validate in image it is pulling?
I can see on https://distribution.github.io/distribution/about/configuration/ that there is a validation configuration, will this do the trick?

**Comment by milosgajdos** [comment]
Please dont use `v2.8.3` release. `v2.8.X` is more or less in maintenance mode at this point. Its not received any updates for a long while and neither will it. Use the latest `v3` release even if it's marked as beta.

**Comment by amir-bialek** [comment]
Hey @milosgajdos 
Thank you for your answer.

Yes I can see ```registry:3.0.0-beta.1``` was updated one month ago.
But are you sure it is ready for use? We are using it for our images and CI, and alto not a production, our CI is dependent on it.

**Comment by milosgajdos** [comment]
Am I sure? No. Never. I never trust software 😄 

Is it better and more stable than `2.8.x`? Heck yeah, but YMMV.

**Comment by amir-bialek** [comment]
Hey @milosgajdos , the error still happening.
one more Q about this topic, when using it as proxy for dockerhub, is this the correct setting for the  remote-url:
```
      REGISTRY_PROXY_REMOTEURL: https://registry.hub.docker.com
```
?

The documentation on docker shows
```
proxy:
  remoteurl: https://registry-1.docker.io
```

**Reply by milosgajdos** [reply]
The `env` var is correct but I am not sure about the URL. Please follow the official Docker docs and use `https://registry-1.docker.io` as your proxy URL.

As for the error, we'd need to get more logs and know what type of manifest it is, what proxy config do you use etc.

**Reply by amir-bialek** [reply]
done. let's see if this will help..
what do you mean by what type of manifest? how can I check this?

**Reply by milosgajdos** [reply]
> what do you mean by what type of manifest? how can I check this?

There are several types of manifests:
* https://distribution.github.io/distribution/spec/manifest-v2-2/
* https://github.com/opencontainers/image-spec/blob/main/manifest.md
* https://github.com/opencontainers/image-spec/blob/main/image-index.md
